### PR TITLE
Create performance resource timing entry for synchronous network fetch

### LINKED ISSUE #23150: Synchronous network fetch does not create performance resource timing entry
All of our async network fetch code ends up calling the `submit_timing` helper function in order to register the resource timing information for the network load. However, the dedicated worker initial script fetch and the worker global importScripts API use the `load_whole_resource` helper API, which blocks until a response is complete and returns the response data. In the handler for `FetchResponseMsg::ProcessResponseEOF` in `load_whole_resource`, the resource timing information is discarded without being used. To fix this, we should:
* make a new SyncFetchResponse structure that contains the resource timing, metadata, and response body
* have `load_whole_resource` return this new structure
* extract the guts of `submit_timing` into a helper `submit_timing_data` function that does not use any generics, but accepts the required arguments
* make `submit_timing` call `submit_timing_data`
* make all callers of `load_whole_resource` call `submit_timing_data` with the required resource timing arguments that are now available

### PR DESCRIPTION
**Pull request by pylbrecht**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #23150 (GitHub issue number if applicable)

- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23322)

### CODE REVIEW

**Comment by pylbrecht** [comment]
@jdm, regarding your idea of passing a function pointer to `load_whole_resource()`:
I'm not sure from where to get the required arguments for `submit_timing_data()` inside `load_whole_resource()`.

Should we just pass more arguments to `load_whole_resource()`?
I see `GlobalScope` and `ServoUrl` floating around in the code calling `load_whole_resource()`, but what about `InitiatorType`?

File: components/script/network_listener.rs [hunk 1]
```diff
@@ -42,14 +44,22 @@ pub fn submit_timing<T: ResourceTimingListener + FetchResponseListener>(listener
...
         url,
         initiator_type,
-        None,
-        &listener.resource_timing(),
+        listener.resource_timing(),
     );
+}
+
+pub fn submit_timing_data(
+    global: &DomRoot<GlobalScope>,
```
Comment (jdm) [inline components/script/network_listener.rs]: This can be `&GlobalScope`.

**Review by jdm — CHANGES_REQUESTED** [review]
Yes, passing more arguments to load_whole_resource is a fine choice. As for InitiatorType, the only callers of load_whole_resource fall into the Other category so we might as well hardcode that. I would expect us to remove these callers over time; the synchronous resource loading API isn't great for slow connections.

**Comment by pylbrecht** [comment]
```
~jdm | pylbrecht: although we may nave a better solution
~jdm | pylbrecht: all of the callers of load_whole_resource live in script/
~jdm | we could just move load_whole_resource to script and make it call submit_timing_data itself              
~jdm | that's going to be more clear and require rebuilding fewer crates in the future
~jdm | load_whole_resource used to be used by lots of other code, but we're slowly getting rid of it            
```

**Review by jdm — APPROVED** [review]

**Pull request merged**